examples/userapi/05_conditional_dimension.ipynb

```diff
@@ -71,7 +71,7 @@
      "name": "stderr",
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #10.    i = Scalar(name='i', dtype=np.int32)
</pre><p>That looks weird, what's the actual issue? Can you open an issue for it?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/devitocodes/devito/pull/1820/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

Reply:
This code, if optimized with openmp+ reduction fails with gcc-5:
```
  START_TIMER(section0)
  for (int x = x_m; x <= x_M; x += 1)
  {
    for (int y = y_m; y <= y_M; y += 1)
    {
      if (f[x][y] != 0)
      {
        int r0 = 1;
        g[1] += r0;
      }
    }
  }
  STOP_TIMER(section0,timers)
```
The trick in the PR helps to generate 

```
  int i = 0;

  /* Begin section0 */
  START_TIMER(section0)
  for (int x = x_m; x <= x_M; x += 1)
  {
    for (int y = y_m; y <= y_M; y += 1)
    {
      if (f[x][y] != 0)
      {
        int r0 = 1;
        g[i + 1] += r0;
      }
    }
  }
  STOP_TIMER(section0,timers)
  /* End section0 */
```
which is valid. Since we are now in a notebook which is not tested for gcc-5, I can get rid of the trick.
Remember that before it was not in notebook, so this was a problem. DO you think it still deserves an issue?

Reply:
That's odd code looks sane enough for GCC what the compilation error?